subprojects/jvm-services/src/main/java/org/gradle/internal/jvm/inspection/JvmInstallationMetadata.java

```diff
@@ -33,8 +33,18 @@ enum JavaInstallationCapability {
         JAVA_COMPILER, J9_VIRTUAL_MACHINE
     }
 
-    static DefaultJvmInstallationMetadata from(File javaHome, String implementationVersion, String runtimeVersion, String jvmVersion, String vendor, String implementationName, String architecture) {
-        return new DefaultJvmInstallationMetadata(javaHome, implementationVersion, runtimeVersion, jvmVersion, vendor, implementationName, architecture);
+    static DefaultJvmInstallationMetadata from(
+        File javaHome,
+        String javaVersion,
```

Comment:
My idea was to keep the names of the fields/parameters as close to the system properties as possible. It is a bit more verbose, but much simpler for the reader to follow. Otherwise, you need to start doing mapping table to understand what is going on 🙂